core/src/muxing.rs

```diff
@@ -241,6 +241,16 @@ impl<T> StreamMuxerEvent<T> {
             None
         }
     }
+
+    /// Map the stream within [`StreamMuxerEvent::InboundSubstream`] to a new type.
+    pub fn map_stream<O>(self, map: impl FnOnce(T) -> O) -> StreamMuxerEvent<O> {
```

Comment:
```suggestion
    pub fn map_inbound_stream<O>(self, map: impl FnOnce(T) -> O) -> StreamMuxerEvent<O> {
```

Just to avoid the confusion of folks expecting the `StreamMuxerEvent` to somehow contain both inbound and outbound streams. What do you think?

Reply:
It will contain both streams once #2648 is merged. If we do this rename now, I will have to rename it again :)

What do you prefer?

Reply:
Though extra work, I would prefer the rename. That way I can merge to `master` already, keeping `master` clean. The two pull requests might not be released in the same release. Mind doing the change (twice)? :innocent: